harness/determined/exec/harness.py

```diff
@@ -111,6 +111,19 @@ def main(train_entrypoint: Optional[str]) -> int:
                 chief_ip=chief_ip,
                 port_offset=info.task_type == "TRIAL" and info.trial._unique_port_offset or 0,
             )
+        elif distributed_backend.use_deepspeed():
+            # World size and rank information set as environment variables in deepspeed launcher.
+            # We pull the relevant fields here and pass them to create the DistributedContext.
+            distributed = _core.DistributedContext(
+                rank=int(cast(str, os.environ.get("RANK"))),
```

Comment:
can these environment variables ever be non-strings? why do we need to cast them?

Reply:
i think the return type of os.environ.get is Optional[str] so that's why we cast as str here for mypy

Reply:
I've said it before, and I'll say it again: you should almost never use `cast`.

In this case, `os.environ.get("RANK")` will not raise an error when `"RANK"` is not in the environment... except that's actually not a supported workflow, so you shouldn't use `.get()`.  The simpler `os.environ["RANK"]` would not require a `cast`.